controlplane/kubeadm/controllers/scale.go

```diff
@@ -63,11 +62,6 @@ func (r *KubeadmControlPlaneReconciler) initializeControlPlane(ctx context.Conte
 func (r *KubeadmControlPlaneReconciler) scaleUpControlPlane(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, controlPlane *internal.ControlPlane) (ctrl.Result, error) {
 	logger := controlPlane.Logger()
 
-	// reconcileHealth returns err if there is a machine being delete which is a required condition to check before scaling up
```

Comment:
Not having doing the health checking/reconciliation here and having the the new check gated on there being no machines in need of remediation seems like we are removing a potential safety guard here.

Reply:
Hmm, what is the scenario where we want to block scaling up due to `reconcileHealth` failing?

Reply:
Any situation where etcd quorum could be lost by adding an additional replica, either briefly in the case that replicas > 2, or permanently if the new instance fails to become an etcd member after it is joined to the cluster.